packages/core/src/tree-view/tree-item.element.scss

```diff
@@ -0,0 +1,135 @@
+
+:host {
+  --animation-duration: #{$cds-global-animation-duration-secondary};
+  --animation-easing: #{$cds-global-animation-easing-primary};
+  --background: #{$cds-alias-object-app-background};
+  --color: #{$cds-global-typography-color-400};
+  --focus-width: #{$cds-alias-object-border-width-300};
+  --font-size: #{$cds-global-typography-font-size-3};
+  --font-weight: #{$cds-global-typography-body-font-weight};
+  cursor: pointer;
+}
+
+.private-host {
+  cursor: inherit;
+
+  .item-children {
+    overflow: hidden;
+  }
+
+  .item-content {
+    background: var(--background);
+    border-left: var(--focus-width) solid transparent;
+    color: var(--color);
+    font-size: var(--font-size);
+    font-weight: var(--font-weight);
+  }
+
+  .item-content:hover {
+    --background: var(--cds-alias-object-interaction-background-hover);
+  }
+
+  .item-content:active {
+    --background: var(--cds-alias-object-interaction-background-active);
+  }
+
+  .lhs-container {
+    width: #{$cds-global-space-7};
+  }
+}
+
+:host(:focus) {
+  outline: none !important;
+
+  .item-content {
+    --background: var(--cds-alias-object-interaction-background-hover);
+    border-color: var(--cds-alias-object-interaction-background-highlight);
+  }
+
+  .item-content:active {
+    --background: var(--cds-alias-object-interaction-background-active);
+  }
+}
+
+:host([disabled]) {
+  .item-content {
+    --color: var(--cds-alias-object-interaction-color-disabled);
+    cursor: not-allowed;
+  }
+
+  .item-content:hover {
+    --background: var(--cds-alias-object-interaction-background-disabled);
+  }
+}
+
+:host([disabled]:focus) {
+  .item-content {
+    --background: var(--cds-alias-object-interaction-background-disabled);
+    border-color: var(--cds-alias-object-interaction-color-disabled);
+  }
+}
+
+:host([selected]) {
+  .item-content {
+    --background: var(--cds-alias-object-interaction-background-selected);
+  }
+}
+
+:host([selected][disabled]) {
+  .item-content {
+    --background: var(--cds-alias-object-interaction-background-disabled);
+  }
+}
+
+:host([cds-motion='off']) {
+  & > .private-host {
+    & > .item-children {
+      height: 0;
+    }
+  }
+}
+
+:host([cds-motion='off'][expanded]) {
+  & > .private-host {
+    & > .item-children {
+      height: auto;
+    }
+  }
+}
+
+:host([cds-motion][_cds-animation-status='ready']:not([cds-motion='off'])) {
+  & > .private-host {
+    & > .item-children {
+      height: 0;
+    }
+  }
+}
+
+:host([cds-motion][expanded][_cds-animation-status='ready']:not([cds-motion='off'])) {
+  & > .private-host {
+    & > .item-children {
+      height: auto !important;
+      transform: none;
+    }
+  }
+}
+
+:host([expanded]) {
+  [part='expand-collapse-icon'] {
+    --icon-transform: rotate(180deg);
+  }
+}
+
+[part='expand-collapse-icon'] {
+  --icon-transform: rotate(90deg);
+  color: var(--cds-alias-object-interaction-color);
+  cursor: inherit;
+  transform: var(--icon-transform);
```

Comment:
I don't know what we're getting by defining a CSS custom property and then immediately using it in the same selector.

I also don't think this is how part is used.

Reply:
Yeah you are right. I was trying to make the component super flexible (allow any custom content projection for the expand/collapse icon) and in the process got confused about how to use the part correctly. 😅 

I've made changes to expose the expand/collapse through part so it can be customized, but got rid of the slot projection ability. App side can do the following to customize the icon and the checkbox (example code available in storybook):

```
.app-custom cds-tree-item::part(checkbox) {
  --color: var(--cds-global-color-jade-700);
  --check-color: var(--cds-global-color-jade-50);
}
.app-custom cds-tree-item::part(expand-collapse-icon) {
  --color: var(--cds-global-color-jade-700);
}
```

Does this seem reasonable? 

Also, I've moved up the `--icon-transform` back to host level. Is this something we should consider exposing at the icon level (e.g. `--transform` property that's consumed by cds-icon)?

Reply:
I like the use of the CSS part API 👍  I think you could get away with not needing `--icon-transform` since the icon is exposed as a part they can apply the transform directly themselves. 

```css
.app-custom cds-tree-item::part(expand-collapse-icon) {
  transform: rotate(180deg);
}
```

Is there any reason to not apply the `expand-collapse-icon` part directly to the `cds-icon` rather than the wrapper div? I guess custom slotted icons would no longer rotate automatically

Reply:
I'm wondering if you are looking at old code, because the part is directly on the `cds-icon`! (It used to not be in my earlier code). I also got rid of the ability to add custom slotted icons for expand / collapse icon.

As for the name of the part, I wanted to keep it specific so as to distinguish from any other projected icons that maybe part of the content of tree node.